doc: Add known issues related to Matter (#19961)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/known_issues.rst

.. rst-class:: v2-9-0 v2-8-0

> KRKNWK-19806: WiFi is not reconnecting to previously connected network after RPU recovery.

Reviewer: ```suggestion
KRKNWK-19806: RPU recovery fails during Wi-Fi communication
   WiFi is not reconnecting to previously connected network after RPU recovery.
```

  **Affected platforms:** nRF7002

>   **Workaround:** Cherry-pick changes from PR #529 in the sdk-connectedhomeip repository.

Reviewer: ```suggestion
  **Workaround:** Cherry-pick changes from PR #529 in the `Matter GitHub repository`_ .
```

.. rst-class:: v2-9-0 v2-8-0

> KRKNWK-19806: RPU recovery fails during Wi-Fi communication

Reviewer 2: Are these issues also present in the v2.9-nRF54H20 branch? 
If so, I can add these issues to PR https://github.com/nrfconnect/sdk-nrf/pull/19745 and add the v2-9-0-nRF54H20-1 tag. This way we can avoid the potential merge conflict during backporting my KI PR to the v2-9-0-nRF54H20 branch.

Author: It's not related to nRF54H20, so I think you can skip it.

>    WiFi is not reconnecting to previously connected network after RPU recovery.

Reviewer: ```suggestion
   Wi-Fi is not reconnecting to previously connected network after RPU recovery.
```

Reviewer: Whoops, forgot this in the previous review.

The issues in this section are related to :ref:`matter_samples`.

.. rst-class:: v2-9-0 v2-8-0

> KRKNWK-19806: RPU recovery fails on nRF5340 DK with nRF7002 EK shield due to invalid MCUboot configuration.

Reviewer: Is this the correct Jira issue? Seems to be the same as the one added above.

Author: Yes, two problems were found related to this issue.

The issues in this section are related to :ref:`matter_samples`.

.. rst-class:: v2-9-0 v2-8-0

> KRKNWK-19806: RPU recovery fails on nRF5340 DK with nRF7002 EK shield due to invalid MCUboot configuration.

Reviewer: ```suggestion
KRKNWK-19806: RPU recovery fails on the nRF5340 DK with nRF7002 EK shield due to invalid MCUboot configuration
   Wi-Fi is not reconnecting to previously connected network after RPU recovery.
```

  **Affected platforms:** nRF5340, nRF7002

>   **Workaround:** Cherry-pick changes from PR #19826 in the sdk-nrf repository.

Reviewer: ```suggestion
  **Workaround:** Cherry-pick changes from PR #19826 in the `sdk-nrf`_ repository.
```
